docs/StardustDocs/topics/operations/multiple/join.md

```diff
@@ -0,0 +1,393 @@
+* `Left` — all rows from left [`DataFrame`](DataFrame.md), mismatches from right [`DataFrame`](DataFrame.md) filled with `null`
+* `Right` — all rows from right [`DataFrame`](DataFrame.md), mismatches from left [`DataFrame`](DataFrame.md) filled with `null`
+* `Full` — all rows from left and right [`DataFrame`](DataFrame.md) objects, any mismatches filled with `null`
+* `Exclude` — only mismatched rows from left [`DataFrame`](DataFrame.md)
+
+For every join type there is a shortcut operation:
+
+```kotlin
+df.innerJoin(otherDf) [ { joinColumns } ]
+df.filterJoin(otherDf) [ { joinColumns } ]
+df.leftJoin(otherDf) [ { joinColumns } ]
+df.rightJoin(otherDf) [ { joinColumns } ]
+df.fullJoin(otherDf) [ { joinColumns } ]
+df.excludeJoin(otherDf) [ { joinColumns } ]
+```
+
+
+### Examples {id="examples_1"}
+
+#### Inner {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_15-->
+
+```kotlin
+// INNER JOIN:
+// Keep only rows where (name, city) match on both sides.
+// In this dataset both Charlies match twice (Moscow, Milan) → 2 merged rows.
+dfLeft.innerJoin(dfRight) { name and city }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_15.html" width="100%" height="500px"></inline-frame>
+
+#### Filter {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_16-->
+
+```kotlin
+// FILTER JOIN:
+// Keep ONLY left rows that have ANY match on (name, city).
+// No right-side columns are added.
+dfLeft.filterJoin(dfRight) { name and city }
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_16.html" width="100%" height="500px"></inline-frame>
+
+#### Left {collapsible="true"}
+
+<!---FUN notebook_test_join_13-->
+
+```kotlin
+dfLeft
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_13.html" width="100%" height="500px"></inline-frame>
+
+<!---FUN notebook_test_join_14-->
+
+```kotlin
+dfRight
+```
+
+<!---END-->
+
+<inline-frame src="./resources/notebook_test_join_14.html" width="100%" height="500px"></inline-frame>
+
+
+<!---FUN notebook_test_join_17-->
+
+```kotlin
+// LEFT JOIN:
+// Keep ALL left rows. If (name, city) matches, attach right columns;
+// if not, right columns are null (e.g., Alice–London has no right match).
```

Comment:
*columns from the right dataframe

Also, Alice-London does have a match, Bob-Dubai does not, so `isBusy == null`